app/src/components/LayoutHeader.vue

```diff
@@ -42,30 +42,138 @@
+
+          <button
+            @click="changeWallet"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            change wallet
+          </button>
+          <button
+            @click="disconnect"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            disconnect wallet
+          </button>
+        </div>
       </div>
-    </nav>
+    </div>
+    <div v-else-if="!isSupportedNetwork" class="ml-auto flex items-center">
+      <div class="flex h-12 items-center pr-2 w-12">
+        <WarningIcon class="icon stroke-pink" />
+      </div>
+      <div class="text-gray-500">Unsupported network</div>
+    </div>
+    <div v-else @click="connectWallet" class="flex items-center h-14 gap-x-2 group cursor-pointer ml-auto">
+      <div class="hidden md:block group-hover:text-grey-500">Connect</div>
+      <div>
+        <ConnectWalletIcon class="icon icon-primary" />
+      </div>
+    </div>
   </header>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 import useWalletStore from 'src/store/wallet';
 import { WarningIcon } from '@fusion-icons/vue/interface';
+import { Wallet3Icon as ConnectWalletIcon } from '@fusion-icons/vue/web3';
+import { ArrowBottom2Icon as ArrowBottomIcon } from '@fusion-icons/vue/interface';
+import Jazzicon from 'src/components/Jazzicon.vue';
+import { Cart2Icon as CartIcon } from '@fusion-icons/vue/interface';
+import { formatEther, JsonRpcProvider, Web3Provider } from 'src/utils/ethers';
 
 // Header menu bar items
 const navigation = [
   { name: 'Home', href: '/' },
   { name: 'Cart', href: '/cart' },
-  { name: 'Contact', href: '/contact' },
 ];
 
 export default defineComponent({
   name: 'LayoutHeader',
-  components: { WarningIcon },
+  components: { ConnectWalletIcon, ArrowBottomIcon, CartIcon, Jazzicon, WarningIcon },
   emits: ['toggleAbout'],
+  data() {
+    return {
+      balance: '0',
+      provider: null as Web3Provider | JsonRpcProvider | null,
+    };
+  },
+  watch: {
+    userAddress: 'updateBalance',
+  },
+  methods: {
+    async updateBalance() {
+      if (!this.userAddress) {
+        return;
+      }
+      const balance = await this.provider?.getBalance?.(this.userAddress);
+      this.balance = formatEther(balance ?? '0');
+    },
+    disconnect() {
+      // ToDo implement disconnect
+      console.log('disconnecting');
+    },
+    changeWallet() {
+      // ToDo implement changeWallet
+      console.log('changing wallets');
+    },
+  },
   setup() {
-    const { connectWallet, isSupportedNetwork, userDisplayName } = useWalletStore();
-    return { connectWallet, isSupportedNetwork, navigation, userDisplayName };
+    const { connectWallet, isSupportedNetwork, provider, userDisplayName, userAddress } = useWalletStore();
+    return { connectWallet, isSupportedNetwork, navigation, provider, userDisplayName, userAddress };
```

Comment:
I'm getting this TS error here. 

TS4082: Default export of the module has or is using private name 'UncheckedJsonRpcSigner'.

![image](https://user-images.githubusercontent.com/2486142/131492754-0ba0c600-55fe-46fb-ac35-485681c92a26.png)

Any ideas?

Reply:
Easy fix: Don't return `provider`! The only things that should be returned from the `setup()` method are things needed in the HTML template, and it seems `provider` is not used in the template (I'd be surprised if you do need it in the HTML template)